src/maker.rs

```diff
@@ -0,0 +1,394 @@
+#![allow(unused_variables)]
+
+use crate::{
+    bitcoin, dai,
+    network::Order,
+    order::{BtcDaiOrder, Position},
+    rate::Spread,
+    MidMarketRate, OngoingTakers, Rate,
+};
+use std::convert::TryFrom;
+
+#[derive(Debug, PartialEq)]
+pub enum NewOrder {
+    Created(BtcDaiOrder),
+}
+
+// Bundles the state of the application
+#[derive(Debug)]
+pub struct Maker {
+    btc_balance: bitcoin::Amount,
+    dai_balance: dai::Amount,
+    btc_fee: bitcoin::Amount,
+    dai_fee: dai::Amount,
+    pub btc_reserved_funds: bitcoin::Amount,
+    pub dai_reserved_funds: dai::Amount,
+    btc_max_sell_amount: bitcoin::Amount,
+    dai_max_sell_amount: dai::Amount,
+    mid_market_rate: MidMarketRate,
+    spread: Spread,
+    ongoing_takers: OngoingTakers,
+}
+
+impl Maker {
+    #![allow(clippy::too_many_arguments)]
+    pub fn new(
+        btc_balance: bitcoin::Amount,
+        dai_balance: dai::Amount,
+        btc_fee: bitcoin::Amount,
+        dai_fee: dai::Amount,
+        btc_max_sell_amount: bitcoin::Amount,
+        dai_max_sell_amount: dai::Amount,
+        mid_market_rate: MidMarketRate,
+        spread: Spread,
+    ) -> Self {
+        Maker {
+            btc_balance,
+            dai_balance,
+            btc_fee,
+            dai_fee,
+            btc_reserved_funds: Default::default(),
+            dai_reserved_funds: Default::default(),
+            btc_max_sell_amount,
+            dai_max_sell_amount,
+            mid_market_rate,
+            spread,
+            ongoing_takers: Default::default(),
+        }
+    }
+
+    pub fn update_rate(&mut self, mid_market_rate: ()) {}
+
+    pub fn update_bitcoin_balance(&mut self, balance: bitcoin::Amount) {
+        self.btc_balance = balance;
+    }
+
+    pub fn update_dai_balance(&mut self, balance: dai::Amount) {
+        self.dai_balance = balance;
+    }
+
+    pub fn track_failed_rate(&mut self, error: anyhow::Error) {}
+
+    pub fn track_failed_balance_update(&mut self, error: anyhow::Error) {}
+
+    pub fn next_sell_order(&self) -> anyhow::Result<BtcDaiOrder> {
```

Comment:
```suggestion
    pub fn sell_order(&self) -> anyhow::Result<BtcDaiOrder> {
```
```suggestion
    pub fn new_sell_order(&self) -> anyhow::Result<BtcDaiOrder> {
```

The `next` terminology is weird, it is not a strict `sequence` of orders.

Reply:
@luckysori do you have strong feelings about this? I'm OK with changing it back to `new` :)

Reply:
I renamed it to `new_...` for now, if we feel strong about these names we can always change them later again.